service/gcs/main.go

```diff
@@ -15,25 +16,44 @@ import (
 )
 
 func main() {
-	// parse command line parameters and init logger
-	if err := utils.ProcessCommandlineOptions(); err != nil {
-		logrus.Fatalf("%+v", err)
+	logLevel := flag.String("loglevel", "warning", "Logging Level: debug, info, warning, error.")
+	logFile := flag.String("logFile", "", "Logging Target: An optional file name/path. Omit for console output.")
+
+	flag.Usage = func() {
+		fmt.Fprintf(os.Stderr, "\nUsage of %s:\n", os.Args[0])
+		flag.PrintDefaults()
+		fmt.Fprintf(os.Stderr, "Examples:\n")
+		fmt.Fprintf(os.Stderr, "    %s -loglevel=debug -logfile=gcslog.txt (default)\n", os.Args[0])
+		fmt.Fprintf(os.Stderr, "    %s -loglevel=info -logfile=stdout\n", os.Args[0])
+	}
+
+	flag.Parse()
+
+	// Use a file instead of stdout
+	if *logFile != "" {
+		logFileHandle, err := os.OpenFile(*logFile, os.O_CREATE|os.O_WRONLY, 0600)
+		if err != nil {
+			logrus.Fatalf("failed to create log file %s", *logFile)
+		}
+		logrus.SetOutput(logFileHandle)
 	}
 
-	// Set logrus output file.
-	// TODO: Consolidate all logs into one file, probably with logrus.
-	const logFilePath = "/tmp/logrus.log"
-	logFile, err := os.OpenFile(logFilePath, os.O_CREATE|os.O_WRONLY, 0600)
+	level, err := logrus.ParseLevel(*logLevel)
 	if err != nil {
-		logrus.Fatalf("%+v", errors.Errorf("failed to create log file %s", logFilePath))
+		logrus.Fatal(err)
 	}
-	logrus.SetOutput(logFile)
 
-	utils.LogMsg("GCS started")
+	if level == logrus.DebugLevel {
+		logrus.AddHook(commonutils.NewStackHook(logrus.AllLevels))
+	}
+
+	logrus.SetLevel(level)
+
+	logrus.Info("GCS started")
 	tport := &transport.VsockTransport{}
 	rtime, err := runc.NewRuntime()
 	if err != nil {
-		logrus.Fatalf("%+v", err)
```

Comment:
This +v was actually fine. We don't need to get rid of all +vs in formatting. We just don't need them for errors which originate from Go standard libraries. the %+v format specifier is really only useful when errors have a stack trace attached, which errors returned from os.Open and other standard library functions don't have. For errors returned from our own internal GCS code, we should always use %+v.